From 093d804611b9a38fe59753b37c29f840518406a9 Mon Sep 17 00:00:00 2001
From: Ken Mills <ken.k.mills@intel.com>
Date: Mon, 13 Dec 2010 15:28:03 +0000
Subject: n_gsm: gsm_data_alloc buffer allocation could fail and it is not being checked

From: Ken Mills <ken.k.mills@intel.com>

commit 093d804611b9a38fe59753b37c29f840518406a9 upstream.

gsm_data_alloc buffer allocation could fail and it is not being checked.

Add check for allocated buffer and return if the buffer allocation
fails.

Signed-off-by: Ken Mills <ken.k.mills@intel.com>
Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/n_gsm.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/char/n_gsm.c
+++ b/drivers/char/n_gsm.c
@@ -968,6 +968,8 @@ static void gsm_control_reply(struct gsm
 {
 	struct gsm_msg *msg;
 	msg = gsm_data_alloc(gsm, 0, dlen + 2, gsm->ftype);
+	if (msg == NULL)
+		return;
 	msg->data[0] = (cmd & 0xFE) << 1 | EA;	/* Clear C/R */
 	msg->data[1] = (dlen << 1) | EA;
 	memcpy(msg->data + 2, data, dlen);
